### ISSUE TITLE
Unable to install package

### ISSUE THREAD

**Issue by heleenbrueggen** [thomvolker/densityratio#9]
Hi Thom, Unfortunately when trying to install the package from Github, I get this warning and the package does not install.

``` r
devtools::install_github("thomvolker/densityratio")
#> Downloading GitHub repo thomvolker/densityratio@HEAD
#> 
#> ── R CMD build ─────────────────────────────────────────────────────────────────
#> * checking for file ‘/private/var/folders/kc/0dq_bqm13h5db_fspf2wz_cc0000gn/T/RtmpccOzCe/remotes27df5aca1654/thomvolker-densityratio-18b8929/DESCRIPTION’ ... OK
#> * preparing ‘densityratio’:
#> * checking DESCRIPTION meta-information ... OK
#> * cleaning src
#> * checking for LF line-endings in source and make files and shell scripts
#> * checking for empty or unneeded directories
#> * building ‘densityratio_0.1.0.tar.gz’
#> Warning in i.p(...): installation of package
#> '/var/folders/kc/0dq_bqm13h5db_fspf2wz_cc0000gn/T//RtmpccOzCe/file27df4ebe0b29/densityratio_0.1.0.tar.gz'
#> had non-zero exit status
```

<sup>Created on 2023-09-22 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by thomvolker** [comment]
The error message does not seem very informative, so a couple of questions:
1. Do you have an Apple or Windows machine? 
2. Do you have a recent (say 4.3.0 or higher) R version?
3. Do you have a recent Rtools installation?
4. Do you have a recent devtools installation?
If the answer to any of 2. to 4. is "no", can you do the update and try again? Let me know if it doesn't work. I will upload a compiled version of the package soon, but I still need to make a couple of updates first.

**Comment by heleenbrueggen** [comment]
I have an Apple computer, my version of R is now 4.3.1 (I've updated it) and I've recently installed the devtools package. When trying to download the package Rtools I get this error: 
``` r
install.packages('Rtools')
#> Warning: package 'Rtools' is not available for this version of R
#> 
#> A version of this package for your version of R might be available elsewhere,
#> see the ideas at
#> https://cran.r-project.org/doc/manuals/r-patched/R-admin.html#Installing-packages
```

<sup>Created on 2023-09-24 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

The warning message when trying to install the package remains the same after updating R:
``` r
devtools::install_github("thomvolker/densityratio")
#> Downloading GitHub repo thomvolker/densityratio@HEAD
#> 
#> ── R CMD build ─────────────────────────────────────────────────────────────────
#> * checking for file ‘/private/var/folders/kc/0dq_bqm13h5db_fspf2wz_cc0000gn/T/RtmpoUohwa/remotes4b9b25084c0c/thomvolker-densityratio-18b8929/DESCRIPTION’ ... OK
#> * preparing ‘densityratio’:
#> * checking DESCRIPTION meta-information ... OK
#> * cleaning src
#> * checking for LF line-endings in source and make files and shell scripts
#> * checking for empty or unneeded directories
#> * building ‘densityratio_0.1.0.tar.gz’
#> Warning in i.p(...): installation of package
#> '/var/folders/kc/0dq_bqm13h5db_fspf2wz_cc0000gn/T//RtmpoUohwa/file4b9b1fa16db2/densityratio_0.1.0.tar.gz'
#> had non-zero exit status
```

<sup>Created on 2023-09-24 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by thomvolker** [comment]
Ah, okay. You will need `Rtools` to install the package, because, unlike packages from CRAN, packages from GitHub need to be build locally, which requires `Rtools`.

I will add a version of `densityratio` to R-universe soon (which is a repository similar to CRAN, but somewhat more low-level). If you cannot get `Rtools` working, please be patient, I'll try to get it on R universe somewhere during the upcoming days/week.

**Comment by heleenbrueggen** [comment]
Seeing that I can only find a version of the Rtools package for Windows and not MacOS, I'm still unable to download the package. I'll just wait then till it is available in the R-universe

**Comment by thomvolker** [comment]
Final thing you could try out for now is following [these steps](https://thecoatlessprofessor.com/programming/cpp/r-compiler-tools-for-rcpp-on-macos/), but otherwise you'll have to wait until the compiled version is online.

**Comment by thomvolker** [comment]
Appeared to be a multi-faceted problem, that is related to (1) not having OpenMP available, which will throw an error during compilation; and (2) not linking to a compiler in R studio. 

The first issue is resolved by commits d29389910eb4fa30bd0947472c1d2e1cf6283369 and 03d46f6e15b122a47c8c45319be5c958f4f13161. The second issue is something a user should resolve, but which on MacOS can be done by following the steps below:

1. Install gcc (including gfortran) by using `brew install gcc
`.

2. Create a file `~/.R/Makevars` (if it does not exist yet).

3. Add the specifications below to `~/.R/Makevars
`

```
FC = /opt/homebrew/Cellar/gcc/13.2.0/bin/gfortran
F77 = /opt/homebrew/Cellar/gcc/13.2.0/bin/gfortran
FLIBS = -L/opt/homebrew/Cellar/gcc/13.2.0/lib/gcc/11
```

**You'll likely have to change the gcc version `13.2.0` to whatever your gcc version is.**

For more information, see [this stackoverflow post](https://stackoverflow.com/questions/69639782/installing-gfortran-on-macbook-with-apple-m1-chip-for-use-in-r).
